packages/data/addon/mirage/handlers/graphql.js

```diff
@@ -62,81 +63,86 @@ function _parseGQLQuery(queryStr) {
-            }
-
-            // Convert each date into a row of data
-            let rows = dates.map(dateTime => ({ dateTime }));
-
-            // Add each dimension
-            columns.dimension.forEach(dimension => {
-              rows = rows.reduce((newRows, currentRow) => {
-                let dimensionValues = _dimensionValues(faker.random.number({ min: 3, max: 5 }));
-
-                return newRows.concat(
-                  dimensionValues.map(value => {
-                    let newRow = Object.assign({}, currentRow);
-                    newRow[dimension] = value;
-                    return newRow;
-                  })
-                );
-              }, []);
-            });
-
-            // Add each metric
-            rows = rows.map(row => {
-              columns.metric.forEach(metric => {
-                row[metric] = faker.finance.amount();
-              });
-
-              return row;
-            });
-
-            return JSON.stringify({
-              data: {
-                [table]: {
-                  edges: rows.map(node => ({ node }))
-                }
-              }
-            });
+function _getResponseBody(db, parent) {
+  // Create mocked response for an async query
+  const { createdOn, query } = parent;
+  const responseTime = Date.now();
+
+  // Only respond if query was created over 5 seconds ago
+  if (responseTime - createdOn >= ASYNC_RESPONSE_DELAY) {
+    parent.status = 'COMPLETE';
+
+    // TODO: get args from _parseGQLQuery result and handle filtering
+    const { table, args, fields } = _parseGQLQuery(JSON.parse(query).query || '');
+
+    if (db.tables.find(table) && fields.length) {
+      const columns = fields.reduce(
+        (groups, column) => {
+          const type = ['metrics', 'dimensions', 'timeDimensions'].find(t => db[t].find(column));
+
+          if (type) {
+            groups[type].push(column);
+          }
+          return groups;
+        },
+        { metrics: [], dimensions: [], timeDimensions: [] }
+      );
+
+      const dates = columns.timeDimensions.length > 0 ? _getDates(args.filter) : [];
+
+      // Convert each date into a row of data
+      // If no time dimension is sent, just return a single row
+      let rows = dates.length ? dates.map(dateTime => ({ dateTime })) : [{}];
+
+      // Add each dimension
+      columns.dimensions.forEach(dimension => {
+        rows = rows.reduce((newRows, currentRow) => {
+          const dimensionValues = _dimensionValues(faker.random.number({ min: 3, max: 5 }));
+
+          return [
+            ...newRows,
+            ...dimensionValues.map(value => ({
+              ...currentRow,
+              [dimension]: value
+            }))
+          ];
+        }, []);
+      });
+
+      // Add each metric
+      rows = rows.map(row => {
+        columns.metrics.forEach(metric => {
+          row[metric] = faker.finance.amount();
+        });
+
+        return row;
+      });
```

Comment:
ah, I meant something like this:
```suggestion
      rows = rows.map((currRow) =>
        columns.metrics.reduce((row, metric) => ({ ...row, [metric]: faker.finance.amount() }), currRow));
```